cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -2425,43 +2414,69 @@ x"
         let e = parse(
             YaccKind::Grmtools,
             "%start A
+                %start A
+                %start A
              %%
              A -> (): 'a1';
-             B -> (): 'bb';
-             A -> (): 'a2';
-             C -> (): 'cc';
-             A -> (): 'a3';
             ",
         )
         .expect_err("parse successful while expecting error");
         assert_eq!(
             e,
             vec![YaccGrammarError {
-                kind: YaccGrammarErrorKind::DuplicateRule,
-                spans: vec![Span::new(38, 39), Span::new(94, 95), Span::new(150, 151)],
+                kind: YaccGrammarErrorKind::DuplicateStartDeclaration,
+                spans: vec![Span::new(7, 8), Span::new(32, 33), Span::new(57, 58)],
             }]
         );
     }
 
     #[test]
-    fn test_duplicate_rules_and_missing_arrow() {
-        let src = "
-        %%
-        A -> () : 'a1';
-        A -> () : 'a2';
-        B -> () : 'b1';
-        A -> () : 'a3';
-        B -> () : 'b2';
-        B";
+    fn test_duplicate_rule() -> Result<(), Vec<YaccGrammarError>> {
```

Comment:
I didn't know test functions could have a non-void return type! Are there any advantages to doing so here?

Reply:
Primarily that you can use `?` (used on line 2443).

Reply:
Is that clearer than `unwrap`? I wonder if it shows you which `?` went wrong?

Reply:
Not sure, I mostly just did it out of habit, i'll have a look at whether there is a difference!

Reply:
It is a little more concise and has less information/no backtrace if I introduce a syntax error.
Result:
```
---- yacc::parser::test::test_duplicate_rule stdout ----
Error: [YaccGrammarError { kind: UnknownDeclaration, spans: [Span { start: 0, end: 0 }] }]
```
unwrap:
```
---- yacc::parser::test::test_duplicate_rule stdout ----
thread 'yacc::parser::test::test_duplicate_rule' panicked at 'called `Result::unwrap()` on an `Err` value: [YaccGrammarError { kind: UnknownDeclaration, spans: [Span { start: 0, end: 0 }] }]', cfgrammar/src/lib/yacc/parser.rs:2423:11
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

Reply:
Interesting! I think `unwrap` "wins" in the sense that if there are two `?`s (not true in this case, I know!), then we can't easily work out which caused the error?